src/lxc/start.c

```diff
@@ -174,6 +174,34 @@ static bool preserve_ns(int ns_fd[LXC_NS_MAX], int clone_flags, pid_t pid)
 	return false;
 }
 
+static int set_oom_score_adj(int oom_score_adj, pid_t pid)
+{
+	int ret;
+	FILE *f;
+/* 5 /proc + 21 /int_as_str + 14 /oom_score_adj + 1 \0 */
+#define OOM_PATH_LEN 45
+	char path[OOM_PATH_LEN];
+
+	ret = snprintf(path, OOM_PATH_LEN, "/proc/%d/oom_score_adj", pid);
+	if (ret < 0 || (size_t)ret >= OOM_PATH_LEN)
+		return -EFBIG;
+
+	f = fopen(path, "w");
+	if (!f) {
+		ERROR("Failed to open %s", path);
+		return -1;
+	}
+
+	if (fprintf(f, "%d", oom_score_adj) < 0) {
```

Comment:
Nit/coding-ocd: Please don't check the function call in the `if`-condition if it's not a boolean. Use
```
ret = fprintf();
if (ret < 0)
       /* handle error */
```